File: core/core-domain/src/main/java/com/coffee/gu/stamp/StampCouponUsage.java
package com.coffee.gu.stamp;

import java.time.LocalDateTime;

public class StampCouponUsage {
    private Long id;
    private Long stampId;
    private Long issuedCouponId;
    private LocalDateTime usedAt;

    public StampCouponUsage(Long id, Long stampId, Long issuedCouponId, LocalDateTime usedAt) {
        this.id = id;
        this.stampId = stampId;
        this.issuedCouponId = issuedCouponId;
        this.usedAt = usedAt;
    }

    public static StampCouponUsage create(Long stampId, Long issuedCouponId, LocalDateTime now) {
        return new StampCouponUsage(null, stampId, issuedCouponId, now);
    }

    public Long getId() {
        return id;
    }

    public Long getStampId() {
        return stampId;
    }

    public Long getIssuedCouponId() {
        return issuedCouponId;
    }

    public LocalDateTime getUsedAt() {
        return usedAt;
    }
}

File: core/core-domain/src/main/kotlin/com/coffee/gu/stamp/StampCouponUsage.kt
package com.coffee.gu.stamp

import java.time.LocalDateTime

class StampCouponUsage(
    val id: Long? = 0,
    val stampId: Long? = 0,
    val issuedCouponId: Long? = 0,
    val usedAt: LocalDateTime = LocalDateTime.now(),
) {
    companion object {
        fun create(
            stampId: Long,
            issuedCouponId: Long,
            now: LocalDateTime,
        ): StampCouponUsage {
            return StampCouponUsage(
                id = 0,
                stampId = stampId,
                issuedCouponId = issuedCouponId,
                usedAt = now,
            )
        }
    }
}
